config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/BundleFileBuilder.java

```diff
@@ -91,12 +91,36 @@ public void buildBundle(File rootDir, File outputDir, List<File> dependencies, S
         }
 
         //Zip
-        final Map<String, Pair<Element, BundleMetadata>> bundleElementMap = bundleEntityBuilder.build(bundle,
-                EntityBuilder.BundleType.DEPLOYMENT, document, projectName, projectGroupName, projectVersion);
-        for (Map.Entry<String, Pair<Element, BundleMetadata>> entry : bundleElementMap.entrySet()) {
-            documentFileUtils.createFile(entry.getValue().getLeft(),
-                    new File(outputDir, entry.getKey() + ".bundle").toPath());
-            jsonFileUtils.createBundleMetadataFile(entry.getValue().getRight(), entry.getKey(), outputDir);
+        final Map<String, Pair<Element, BundleMetadata>> annotatedElements = new LinkedHashMap<>();
+        final AnnotatedEntityCreator annotatedEntityCreator = AnnotatedEntityCreator.INSTANCE;
+        // Filter the bundle to export only annotated entities
+        // TODO : Enhance this logic to support services and policies
+        bundle.getEntities(Encass.class).values().stream()
+                .filter(Encass::hasAnnotated)
+                .map(encass -> annotatedEntityCreator.createAnnotatedEntity(encass, projectName, projectVersion))
+                .forEach(annotatedEntity -> {
+                    if (annotatedEntity.isBundleTypeEnabled()) {
+                        // buildEncassDependencies
+                        final Pair<Element, BundleMetadata> bundleMetadataPair = bundleEntityBuilder.build(bundle,
+                                EntityBuilder.BundleType.DEPLOYMENT, document, projectName, projectGroupName, projectVersion, annotatedEntity);
+                        if(bundleMetadataPair != null){
+                            annotatedElements.put(annotatedEntity.getBundleName(), bundleMetadataPair);
+                        }
+                    }
+                });
+
+        if(annotatedElements.isEmpty()) {
+            annotatedElements.put(StringUtils.isBlank(projectVersion) ? projectName : projectName + "-" + projectVersion, bundleEntityBuilder.build(bundle,
+                    EntityBuilder.BundleType.DEPLOYMENT, document, projectName, projectGroupName, projectVersion, null));
+        }
+
+        for (Map.Entry<String, Pair<Element, BundleMetadata>> entry : annotatedElements.entrySet()) {
+            Pair<Element, BundleMetadata> elementBundleMetadataPair = entry.getValue();
+            if(elementBundleMetadataPair != null) {
+                documentFileUtils.createFile(elementBundleMetadataPair.getLeft(),
+                        new File(outputDir, entry.getKey() + ".bundle").toPath());
+                jsonFileUtils.createBundleMetadataFile(entry.getValue().getRight(), entry.getKey(), outputDir);
```

Comment:
We should have a null check for `BundleMetadata`. This should have been done in PR #212. Can you take care of it?